probe-rs/src/probe/daplink/mod.rs

```diff
@@ -318,25 +310,23 @@ impl DAPAccess for DAPLink {
             Port::AccessPort(_) => PortType::AP,
         };
 
-        commands::send_command::<TransferRequest, TransferResponse>(
+        let response = commands::send_command::<TransferRequest, TransferResponse>(
             &mut self.device,
             TransferRequest::new(InnerTransferRequest::new(port, RW::W, addr as u8), value),
-        )
-        .map_err(|_| DebugProbeError::UnknownError)
-        .and_then(|v| {
-            if v.transfer_count == 1 {
-                if v.transfer_response.protocol_error {
-                    Err(DebugProbeError::USBError(None))
-                } else {
-                    match v.transfer_response.ack {
-                        Ack::Ok => Ok(()),
-                        _ => Err(DebugProbeError::UnknownError),
-                    }
-                }
+        )?;
+
+        if response.transfer_count == 1 {
+            if response.transfer_response.protocol_error {
+                Err(DebugProbeError::USBError(None))
```

Comment:
```suggestion
                Err(DebugProbeError::USB(None))
```

Reply:
Even tho this was not introduced in this PR, we could change this whilst we are at it.